src/logger.rs

```diff
@@ -0,0 +1,47 @@
+//! Module containing logger implementation.
+
+use crate::ffi::string::{AscStr, AscString};
+use log::{Level, LevelFilter, Log, Metadata, Record};
+
+/// The main logger implementation for the `log` facade crate.
+pub struct Logger;
+
+impl Log for Logger {
+    fn enabled(&self, _: &Metadata) -> bool {
+        true
+    }
+
+    fn log(&self, record: &Record) {
+        let level = match record.level() {
+            Level::Error => ERROR,
+            Level::Warn => WARNING,
+            Level::Info => INFO,
+            Level::Debug | Level::Trace => DEBUG,
+        };
+        let message = AscString::new(record.args().to_string());
+
+        unsafe {
+            log(level, &message);
```

Comment:
So this gets resolved into the logger from the `extern` block, right? Why do we need to use that log function in C? Instead of just the log directly?

Reply:
`log` is just a facade that calls the logger implementation that we initialize on startup. This is why we need a `start` function that calls `logger::init()` and registers the logger with the `log` crate.